src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -33,95 +34,76 @@
+        } else {
+            this.kafkaConfigurationMap = new HashMap<>();
+            this.kafkaConfigurationMap.putAll(builder.kafkaConfigurationMap);
+        }
+
+        LOGGER.info("Set broker.id to: {}", this.brokerId);
         this.kafkaConfigurationMap.put("broker.id", String.valueOf(this.brokerId));
-    }
 
-    /**
-     * Create @code{StrimziKafkaContainer} instance with external ZooKeeper (i.e., @code{StrimziZooKeeperContainer}).
-     *
-     * Creating the container using this method disables Kraft.
-     *
-     * @param brokerId broker id
-     * @param connectString connect string
-     * @param additionalKafkaConfiguration addtional configuration
-     * @return instance of @code{StrimziKafkaContainer}
-     */
-    public static StrimziKafkaContainer createWithExternalZookeeper(final int brokerId,
-                                                                    final String connectString, final Map<String, String> additionalKafkaConfiguration) {
-        return new StrimziKafkaContainer(brokerId, false, additionalKafkaConfiguration)
-            .withExternalZookeeper(connectString);
-    }
+        LOGGER.info("Injected configuration inside Kafka Container constructor....\n{}", this.kafkaConfigurationMap.toString());
 
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @param useKraft whether the broker should use Kraft
-     * @param additionalKafkaConfiguration additional configuration for Kafka broker
-     * @return instance of @code{StrimziKafkaContainer}
-     */
-    public static StrimziKafkaContainer createWithAdditionalConfiguration(final int brokerId, final boolean useKraft, final Map<String, String> additionalKafkaConfiguration) {
-        return new StrimziKafkaContainer(brokerId, useKraft, additionalKafkaConfiguration);
-    }
+        this.externalZookeeperConnect = builder.externalZookeeperConnect;
 
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @param additionalKafkaConfiguration additional configuration for Kafka broker
-     * @return instance of @code{StrimziKafkaContainer}
-     */
-    public static StrimziKafkaContainer createWithAdditionalConfiguration(final int brokerId, final Map<String, String> additionalKafkaConfiguration) {
-        return createWithAdditionalConfiguration(brokerId, false, additionalKafkaConfiguration);
+        // we need this shared network in case we deploy StrimziKafkaCluster which consist of `StrimziKafkaContainer`
+        // instances and by default each container has its own network, which results in `Unable to resolve address: zookeeper:2181`
+        this.withNetwork(Network.SHARED);
+        // exposing kafka port from the container
+        this.withExposedPorts(KAFKA_PORT);
+        this.withEnv("LOG_DIR", "/tmp");
```

Comment:
This is a bit weird since the `with*` methods are normally part of the builder pattern. Should these be just `set*` methods? It seems like we are mixing the things around.

Reply:
I have changed it to set methods...(but those methods basically encapsulate with methods)